nusight2/src/client/components/localisation/view.tsx

```diff
@@ -201,3 +202,19 @@ const FieldLineDots = ({ model }: { model: LocalisationModel }) => (
     )}
   </>
 );
+
+const Ball = ({ model }: { model: LocalisationModel }) => (
+  <>
+    {model.robots.map(
+      (robot) =>
+        robot.visible && (
+          <object3D key={robot.id}>
+            <mesh position={robot.ball.toArray()}>
+              <sphereBufferGeometry args={[robot.ball.z, 20, 20]} />
```

Comment:
It is expensive to have a dynamic geometry definition, it is more efficient to instead scale the mesh dynamically:

```suggestion
            <mesh position={robot.ball.toArray()} scale={[robot.ball.z, robot.ball.z, robot.ball.z]}>
              <sphereBufferGeometry args={[1, 20, 20]} />
```

Although, why exactly are we scaling by `ball.z`?

Reply:
That would make it so the ball is always on the ground and just grows to do so?
I'm not sure that's what was intended but I think that's what it would do